cli_tools/common/utils/logging/service/cli_tool_with_log.go

```diff
@@ -0,0 +1,44 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package service
+
+import (
+	"flag"
+	"os"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+// CliToolWithLogging abstracts the interface of a cli tool with logging
+type CliToolWithLogging interface {
```

Comment:
There's nothing specific to logging in this interface. It should be in a package common to CLI tools in general, not in a logging package. 

You can put UpdatedParams struct there as well (the other comment about not using map[string]string for updated params)

Reply:
It does have specific logic for logging. It defines some methods, which serves the function splitting of logging. RunCliToolWithLogging Executes the main flow of a cli tool with logging.

Correspondingly, core functionality of those tools can also be executed directly without this wrap. Directly call "Run" and no logging will happen.

There are also some other cli tools, e.g. gce_image_publish, who doesn't implement this interface. So, those tools won't have logging feature naturally.